### PR TITLE
drivers: si70xx: adapt to new I2C api.

### PR DESCRIPTION
**Pull request by basilfx**
### Contribution description

This PR updates the Si70xx drivers. Tested this one on the SLTB001A (EFM32, adapted).

### Issues/PRs references

#6577

### CODE REVIEW

File: drivers/si70xx/si70xx.c [hunk 1]
```diff
@@ -39,11 +39,11 @@ static uint16_t _do_measure(const si70xx_t *dev, uint8_t command)
 
     i2c_acquire(SI70XX_I2C);
 
-    if (i2c_write_byte(SI70XX_I2C, SI70XX_ADDR, command) != 1) {
```
Comment (dylad) [inline drivers/si70xx/si70xx.c]: I2C_ACK doesn't exist anymore as we switch to errno used. This must be replaced by 0.

**Comment by basilfx** [comment]
Rebased and adapted.

**Comment by basilfx** [comment]
Squashed.

**Comment by dylad** [comment]
Thanks for your work @basilfx ! Here we go !

**Pull request merged**